File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while True:
            prod = 1
            temp = n
            while temp > 0:
                prod *= temp % 10
                temp //= 10
            if prod % t == 0:
                return n
            n += 1

File: LeetCode/3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        while (true) {
            int prod = 1;
            int temp = n;
            while (temp) {
                prod *= temp % 10;
                temp /= 10;
            }
            if (prod % t == 0) return n;
            n++;
        }
        return 0;
    }
};